Sofa/GL/Component/Shader/src/sofa/gl/component/shader/LightManager.cpp

```diff
@@ -490,26 +490,9 @@ void LightManager::handleEvent(sofa::core::objectmodel::Event* event)
 
         case 'l':
         case 'L':
-            if (!m_shadowShaders.empty())
-            {
-                const bool b = d_shadowsEnabled.getValue();
-                d_shadowsEnabled.setValue(!b);
-                if (!m_shadowShaders.empty())
-                {
-                    for (unsigned int i=0 ; i < m_shadowShaders.size() ; ++i)
-                    {
-                        m_shadowShaders[i]->setCurrentIndex(d_shadowsEnabled.getValue() ? 1 : 0);
-                        m_shadowShaders[i]->updateVisual();
-                    }
-                    for (std::vector<Light::SPtr>::iterator itl = m_lights.begin(); itl != m_lights.end() ; ++itl)
-                    {
-                        (*itl)->updateVisual();
-                    }
-                    this->updateVisual();
-                }
-
-                msg_info() << "Shadows : "<<(d_shadowsEnabled.getValue()?"ENABLED":"DISABLED");
-            }
+            msg_deprecated() << "This component was implemented with hard coded user interactions. This behavior was deprecated in PR#4163. "
+                                "If you need the old behavior, you need to re-implement with a controller. This can be done"
+                                " c++ or python. If you see this message  and are unable implement a controller on your own, please report at https://github.com/sofa-framework/sofa/pull/4163";
```

Comment:
```suggestion
                                " c++ or python. If you see this message and are unable to implement a controller on your own, please report at https://github.com/sofa-framework/sofa/pull/4163";
```
to apply on all messages (or in a common function?).